signer/src/transaction_signer.rs

```diff
@@ -404,7 +497,7 @@ mod tests {
                 .unwrap()
             {
                 if context_window_block_hashes.contains(&deposit_request_block) {
-                    assert_eq!(signer_decisions.len(), 1);
+                    assert_eq!(signer_decisions.len(), num_expected_decisions);
                     assert!(signer_decisions.first().unwrap().is_accepted)
                 } else {
                     assert_eq!(signer_decisions.len(), 0);
```

Comment:
For any block that isn't part of the context window, there shouldn't be any decisions in this scenario - because the transaction signer will only observe transactions within the context window.